management/server/idp/zitadel.go

```diff
@@ -0,0 +1,605 @@
+package idp
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/golang-jwt/jwt"
+	"github.com/netbirdio/netbird/management/server/telemetry"
+	log "github.com/sirupsen/logrus"
+)
+
+// ZitadelManager zitadel manager client instance.
+type ZitadelManager struct {
+	managementEndpoint string
+	httpClient         ManagerHTTPClient
+	credentials        ManagerCredentials
+	helper             ManagerHelper
+	appMetrics         telemetry.AppMetrics
+}
+
+// ZitadelClientConfig zitadel manager client configurations.
+type ZitadelClientConfig struct {
+	ClientID           string
+	ClientSecret       string
+	GrantType          string
+	TokenEndpoint      string
```

Comment:
I think that we can pre-load this endpoint automatically from the OIDC config here:

https://github.com/netbirdio/netbird/blob/fea53b2f0ff97485ef5c1ad9090d7afbf6747f76/management/cmd/management.go#L371

Reply:
Also the user won't need to provide it in the docs.